jina/peapods/zmq.py

```diff
@@ -313,6 +313,8 @@ def register_pollin(self):
     def close(self):
         """Close all sockets and shutdown the ZMQ context associated to this `Zmqlet`. """
         if not self.is_closed:
+            # wait until the close signal is received
+            time.sleep(.01)
```

Comment:
isn't there a better way to ensure this?

Reply:
nah, not for now.